src/components/FeatureViews/AdminPortal/UnpaidSessions.js

```diff
@@ -12,23 +15,39 @@ import "./AdminPortal.scss";
 
 const UnpaidSessions = () => {
 
-	const UnpaidList = useSelector(({Admin}) => Admin.Unpaid) || [];
-	const studentList = useMemo(() => UnpaidList.map(({student}) => student), [
-		UnpaidList,
-	]);
-	const courseList = useMemo(() => UnpaidList.map(({course}) => course), [
-		UnpaidList,
-	]);
+	const QUERIES = {
+		"unpaidSessions": gql`query MyQuery {
+			unpaidSessions {
+			  student {
+				user {
+				  id
+				  firstName
+				  lastName
+				}
+			  }
+			  course {
+				id
+				title
+				startTime
+				endTime
+				hourlyTuition
+			  }
+			  sessionsLeft
+			  lastPaidSessionDatetime
+			}
+		  }
+		  `
+	}
 
-	const studentStatus = hooks.useStudent(studentList);
-	const courseStatus = hooks.useCourse(courseList);
-	const unpaidSessionStatus = hooks.useUnpaidSessions();
+	const { data, loading, error } = useQuery(QUERIES["unpaidSessions"]);
 
-	if (hooks.isLoading(studentStatus, courseStatus, unpaidSessionStatus) && (courseStatus !== null && studentStatus !== null)) {
+	if (loading){
 		return <Loading loadingText="UNPAID SESSIONS LOADING" small/>;
 	}
```

Comment:
Add error handling:
```
if(error) => display `error.message`
```

Reply:
Is the message "There has been an error" plus console.error sufficiently descriptive?

Reply:
"There has been an error loading unpaid sessions"?

Reply:
the error object has a message property so display something like `<div> There has been an error: {error.message} </div>`